Adjust girq_regs to avoid flexible-array-like behaviors

Snape3058 opened the pull request:
Struct girq_regs has an array field of length 1 at its end. Since this field is never used in the code base, we can replace it with a non-array field of its original element type to remove the flexible-array-like behaviors and avoid using a flexible array file in the middle of a struct or defining an array of flexible array struct type.

Fixes #84251

Snape3058 commented:
I have tried to do some local tests on this change. But there will still be some places that I cannot cover. Wait for the CI results and apologize in advance in case it fails.

Snape3058 commented:
It seems that all tests pass for this change. Please help me update the commit message if it looks good to merge, or provide me with other suggestions for further updates. Thanks.

albertofloyd commented:
> It seems that all tests pass for this change. Please help me update the commit message if it looks good to merge, or provide me with other suggestions for further updates. Thanks.

Do change commit message as indicated in CI error.
e.g. 
soc: mec172x: ecia: Adjust girq_regs to avoid flexible-array-like behaviors

Snape3058 committed:
soc: mec172x: ecia: Adjust girq_regs to avoid flexible-array-like behaviors

Struct girq_regs has an array field of length 1 at its end. Since this
field is never used in the code base, we can replace it with a non-array
field of its original element type to remove the flexible-array-like
behaviors and avoid using a flexible array file in the middle of a
struct or defining an array of flexible array struct type.

Fixes zephyrproject-rtos#84251

Signed-off-by: Ella MA <xutong.ma@inria.fr>

Snape3058 commented:
Hi, how is this patch going? Do I need to take any action with it? thx

Snape3058 commented:
Thank you. Can we merge now?
